gimmik/__init__.py

```diff
@@ -30,8 +32,16 @@ def generate_mm(mat, dtype, platform, alpha=1.0, beta=0.0, funcn='gimmik_mm'):
     src = Template(tpl).render(**tplargs)
 
     # At single precision suffix all floating point constants by 'f'
-    if dtype == 'float':
-        src = re.sub(r'(?=\d*[.eE])(?=\.?\d)\d*\.?\d*(?:[eE][+-]?\d+)?',
+    if 'f90' in platform :
+        if 'kind=4' in dtype:
+            src = re.sub(r'(?=\d*[.eE])(?=\.?\d)\d*\.?\d*(?:[eE][+-]?\d+)?',
+                     r'\g<0>_4', src)
```

Comment:
Factor out the subst bit into a variable so there is only one re.sub call for all three cases.

Reply:
substs = {'float': 'f', 'real(kind=4)': '_4', 'real(kind=8)': '_4'}
if dtype in substs:
    src = re.sub(..., r'\g<0>' + substs[dtype], src)